### ISSUE TITLE
`root` with empty password security vulnerability. How to change the default password?

### ISSUE THREAD

**Issue by Osiris-Team** [MariaDB4j/MariaDB4j#819]
It seems to me that the database `root` user with an empty password is a pretty big security vulnerability,
since many Linux distributions come with all ports open. Not only that but also other programs on the same machine can access the database easily if the password isn't changed (how to change the password programmatically btw?).

Thus I'd like to generate a random password by default for the root user instead of an empty password which then can be got programmatically like so:

```java
Connection conn = DriverManager.getConnection(configBuilder.getURL(dbName), "root", configBuilder.getPassword("root"));
```

Otherwise being able to change the password for the root user programmatically via for example `db.setRootPassword("pass")` would be also great.

**Comment by vorburger** [comment]
@Osiris-Team or anyone else reading along here: Pull Requests proposing to address this are most welcome!

If you would like me to make this fix for you, then please (README quote) _support [on OpenCollective](https://opencollective.com/mariadb4j), via [GitHub Sponsoring](https://github.com/sponsors/vorburger) or through [a Tidelift subscription](https://tidelift.com)** to ensure active maintenance of this project [used by hundreds](https://github.com/MariaDB4j/MariaDB4j/network/dependents), since 2011! 🫶_

**Comment by vorburger** [comment]
I do agree that it would be good to review and change this. I do welcome PRs for this.

In the mean time,  see the `MariaDB4jSampleTutorialTest.testEmbeddedMariaDB4jWithSecurity()` for how to programmatically set a password.

**Comment by Osiris-Team** [comment]
@vorburger is the generated password reusable between restarts? Meaning does it always generate the same random password?

**Comment by vorburger** [comment]
> [@vorburger](https://github.com/vorburger) is the generated password reusable between restarts? Meaning does it always generate the same random password?

How do you mean? Always generating the same random password wouldn't be very... random, would it? 🤣

But I actually do know what you really mean, and thought about it as well. The only way that I could see this work would be that:

a) If you use MariaDB4j for ephemeral databases in tests, it generates a random password (it being MariaDB4j core itself, not the test code, as currently), the test uses a (MariaDB4j) API to get that password - and uses it for connecting. It's not persistent, and the next time the test re-runs it changes - which is perfectly fine, all is well. 

b) if you use MariaDB4j to create a DB for some sort of "embedded" kinda use case, and need to reconnect to a DB that's in a "non-`/tmp`" sort of data directory across JVM restarts, then the application code using MariaDB4j is responsible for externally persisting that "secret" password 🔑 ... somewhere, somehow. How it really does that is probably not something that this library should be involved in.

Unless I'm missing something obvious (do share ideas!), it seems to me that's the only way to do this...

Does this make sense to you as well?

**Comment by Osiris-Team** [comment]
Yeah for me it's the second use case, okay then thanks for the info and the example code in the test! Will make sure to use it.

**Comment by Osiris-Team** [comment]
Yeah, makes sense to leave this in the application code and responsibility. Here is the example: https://github.com/MariaDB4j/MariaDB4j/blob/5b423bb691293c85c6735eac77a5b74af62cae36/mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java#L144
